be/src/vec/data_types/serde/data_type_array_serde.cpp

```diff
@@ -299,18 +299,28 @@ void DataTypeArraySerDe::read_column_from_arrow(IColumn& column, const arrow::Ar
 template <bool is_binary_format>
 Status DataTypeArraySerDe::_write_column_to_mysql(const IColumn& column,
                                                   MysqlRowBuffer<is_binary_format>& result,
-                                                  int row_idx, bool col_const) const {
+                                                  int row_idx_of_mysql, bool col_const) const {
     auto& column_array = assert_cast<const ColumnArray&>(column);
     auto& offsets = column_array.get_offsets();
     auto& data = column_array.get_data();
     bool is_nested_string = data.is_column_string();
-    const auto col_index = index_check_const(row_idx, col_const);
+    const auto row_idx_of_col_arr = index_check_const(row_idx_of_mysql, col_const);
+
+    if (column_array.size() <= row_idx_of_col_arr) {
```

Comment:
I do not think here should add this check, if here add , string serde should add same check too.

Reply:
Yes, we should add check for all serde if we can.

Reply:
if u add array , do same with map too.

Reply:
check is removed since it is checked per row.
this pr just fixed mysql result writer, and we have other writer types, maybe they are subjected to the same problem